app/lib/three_scale/api/collection.rb

```diff
@@ -36,7 +38,21 @@ def respond_to_missing?(method_name, include_private = false)
 
   private
 
-  def pagination
-    pagination_attrs(@collection)
+  attr_reader :collection
+
+  def represented
+    @represented ||= collection.try(:represented) || collection
+  end
+
+  def pagination_attrs(wrapper: nil)
+    return {} if represented.per_page >= represented.total_entries
```

Comment:
I think this checks belongs a level higher at `#to_json`
```suggestion
```

```rb
def to_json(options = {})
  MultiJson.dump collection.to_hash(options)
  collection.merge(pagination_attrs(wrapper: :metadata)) if paginated?
end

# This might seem a bit overdone but it has a slightly better readability. It may look
# odd at first to compare per_page with total_entries
def paginated?
  represented.per_page >= represented.total_entries
end
```

Reply:
then it would be like this:

```ruby
def to_json(options = {})
  MultiJson.dump collection.to_hash(options)
  collection.merge(pagination_attrs(wrapper: :metadata)) if paginated?
end

private



def paginated?
  represented.per_page >= represented.total_entries
rescue NoMethodError
  false
end
```